File: app/src/main/java/pl/pola_app/ui/activity/MenuActivity.java
package pl.pola_app.ui.activity;

import android.content.ActivityNotFoundException;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import pl.pola_app.BuildConfig;
import pl.pola_app.R;
import pl.pola_app.databinding.ActivityMenuBinding;
import pl.pola_app.helpers.EventLogger;
import pl.pola_app.helpers.SessionId;
import pl.pola_app.helpers.Utils;

public class MenuActivity extends AppCompatActivity {

    private ActivityMenuBinding binding;

    private EventLogger logger;
    private SessionId sessionId;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMenuBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        logger = new EventLogger(this);
        sessionId = SessionId.create(this);
        setupView();
        binding.menuBackIv.setOnClickListener(this::onMenuBackClick);
        binding.activityMenuAboutAppTv.setOnClickListener(this::onAboutAppClick);
        binding.activityMenuUserManualTv.setOnClickListener(this::onUserManualClick);
        binding.activityMenuAboutKjTv.setOnClickListener(this::onAboutKJClick);
        binding.activityMenuTeamTv.setOnClickListener(this::onTeamClick);
        binding.activityMenuPartnersTv.setOnClickListener(this::onPartnersClick);
        binding.activityMenuFriendsTv.setOnClickListener(this::onFriendsClick);
        binding.activityFoundBugTv.setOnClickListener(this::onFoundBugClick);
        binding.activityRateTv.setOnClickListener(this::onRateClick);
        binding.activityFacebookTv.setOnClickListener(this::onFacebookClick);
        binding.activityTwitterTv.setOnClickListener(this::onTwitterClick);
    }

    private void setupView() {
        binding.appBuildTv.setText(getString(R.string.pola_application, BuildConfig.VERSION_NAME, BuildConfig.VERSION_CODE));
    }

    void onMenuBackClick(View view) {
        finish();
    }

    void onAboutAppClick(View view) {
        logger.logMenuItemOpened("O Aplikacji Pola", sessionId.get());
        Intent intent = new Intent(this, ActivityWebView.class);
        intent.putExtra("url", Utils.URL_POLA_ABOUT);
        startActivity(intent);
    }

    void onUserManualClick(View view) {
        logger.logMenuItemOpened("Instrukcja obsługi", sessionId.get());
        Intent intent = new Intent(this, ActivityWebView.class);
        intent.putExtra("url", Utils.URL_POLA_METHOD);
        startActivity(intent);
    }

    void onAboutKJClick(View view) {
        logger.logMenuItemOpened("O Klubie Jagiellońskim", sessionId.get());
        Intent intent = new Intent(this, ActivityWebView.class);
        intent.putExtra("url", Utils.URL_POLA_KJ);
        startActivity(intent);
    }

    void onTeamClick(View view) {
        logger.logMenuItemOpened("Zespół", sessionId.get());
        Intent intent = new Intent(this, ActivityWebView.class);
        intent.putExtra("url", Utils.URL_POLA_TEAM);
        startActivity(intent);
    }

    void onPartnersClick(View view) {
        logger.logMenuItemOpened("Partnerzy", sessionId.get());
        Intent intent = new Intent(this, ActivityWebView.class);
        intent.putExtra("url", Utils.URL_POLA_PARTNERS);
        startActivity(intent);
    }

    void onFriendsClick(View view) {
        logger.logMenuItemOpened("Przyjaciele Poli", sessionId.get());
        Intent intent = new Intent(this, ActivityWebView.class);
        intent.putExtra("url", Utils.URL_POLA_FRIENDS);
        startActivity(intent);
    }

    void onFoundBugClick(View view) {
        logger.logMenuItemOpened("Zgłoś błąd w danych", sessionId.get());
        Intent intent = new Intent(this, CreateReportActivity.class);
        intent.setAction("product_report");
        startActivity(intent);
    }

    void onRateClick(View view) {
        logger.logMenuItemOpened("Pola na Twitterze", sessionId.get());
        Uri uri = Uri.parse("market://details?id=" + this.getPackageName());
        Intent intent = new Intent(Intent.ACTION_VIEW, uri);
        intent.addFlags(Intent.FLAG_ACTIVITY_NO_HISTORY |
                Intent.FLAG_ACTIVITY_CLEAR_WHEN_TASK_RESET |
                Intent.FLAG_ACTIVITY_MULTIPLE_TASK);
        try {
            startActivity(intent);
        } catch (ActivityNotFoundException e) {
            startActivity(new Intent(Intent.ACTION_VIEW,
                    Uri.parse("http://play.google.com/store/apps/details?id=" + this.getPackageName())));
        }
    }

    void onFacebookClick(View view) {
        logger.logMenuItemOpened("Pola na Facebooku", sessionId.get());
        startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse(Utils.URL_POLA_FB)));
    }

    void onTwitterClick(View view) {
        logger.logMenuItemOpened("Pola na Twitterze", sessionId.get());
        startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse(Utils.URL_POLA_TWITTER)));
    }
}

File: app/src/main/java/pl/pola_app/ui/activity/MenuActivity.kt
package pl.pola_app.ui.activity

import android.content.ActivityNotFoundException
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import pl.pola_app.BuildConfig
import pl.pola_app.R.string
import pl.pola_app.databinding.ActivityMenuBinding
import pl.pola_app.helpers.EventLogger
import pl.pola_app.helpers.SessionId
import pl.pola_app.helpers.Utils

class MenuActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMenuBinding
    private lateinit var logger: EventLogger
    private lateinit var sessionId: SessionId

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMenuBinding.inflate(layoutInflater)
        setContentView(binding.root)
        logger = EventLogger(this)
        sessionId = SessionId.create(this)
        setupView()
        with(binding) {
            menuBackIv.setOnClickListener { onMenuBackClick() }
            activityMenuAboutAppTv.setOnClickListener { onAboutAppClick() }
            activityMenuUserManualTv.setOnClickListener { onUserManualClick() }
            activityMenuAboutKjTv.setOnClickListener { onAboutKlubJagiellonskiClick() }
            activityMenuTeamTv.setOnClickListener { onTeamClick() }
            activityMenuPartnersTv.setOnClickListener { onPartnersClick() }
            activityMenuFriendsTv.setOnClickListener { onFriendsClick() }
            activityFoundBugTv.setOnClickListener { onFoundBugClick() }
            activityRateTv.setOnClickListener { onRateClick() }
            activityGhTv.setOnClickListener { onGhClick() }
            activityFacebookTv.setOnClickListener { onFacebookClick() }
            activityTwitterTv.setOnClickListener { onTwitterClick() }
        }
    }

    private fun setupView() {
        binding.appBuildTv.text = getString(string.pola_application, BuildConfig.VERSION_NAME, BuildConfig.VERSION_CODE)
    }

    private fun onMenuBackClick() {
        finish()
    }

    private fun onAboutAppClick() {
        logger.logMenuItemOpened("O Aplikacji Pola", sessionId.get())
        val intent = Intent(this, ActivityWebView::class.java)
        intent.putExtra("url", Utils.URL_POLA_ABOUT)
        startActivity(intent)
    }

    private fun onUserManualClick() {
        logger.logMenuItemOpened("Instrukcja obsługi", sessionId.get())
        val intent = Intent(this, ActivityWebView::class.java)
        intent.putExtra("url", Utils.URL_POLA_METHOD)
        startActivity(intent)
    }

    private fun onAboutKlubJagiellonskiClick() {
        logger.logMenuItemOpened("O Klubie Jagiellońskim", sessionId.get())
        val intent = Intent(this, ActivityWebView::class.java)
        intent.putExtra("url", Utils.URL_POLA_KJ)
        startActivity(intent)
    }

    private fun onTeamClick() {
        logger.logMenuItemOpened("Zespół", sessionId.get())
        val intent = Intent(this, ActivityWebView::class.java)
        intent.putExtra("url", Utils.URL_POLA_TEAM)
        startActivity(intent)
    }

    private fun onPartnersClick() {
        logger.logMenuItemOpened("Partnerzy", sessionId.get())
        val intent = Intent(this, ActivityWebView::class.java)
        intent.putExtra("url", Utils.URL_POLA_PARTNERS)
        startActivity(intent)
    }

    private fun onFriendsClick() {
        logger.logMenuItemOpened("Przyjaciele Poli", sessionId.get())
        val intent = Intent(this, ActivityWebView::class.java)
        intent.putExtra("url", Utils.URL_POLA_FRIENDS)
        startActivity(intent)
    }

    private fun onFoundBugClick() {
        logger.logMenuItemOpened("Zgłoś błąd w danych", sessionId.get())
        val intent = Intent(this, CreateReportActivity::class.java)
        intent.action = "product_report"
        startActivity(intent)
    }

    private fun onRateClick() {
        logger.logMenuItemOpened("Pola na Twitterze", sessionId.get())
        val uri = Uri.parse("market://details?id=" + this.packageName)
        val intent = Intent(Intent.ACTION_VIEW, uri)
        intent.addFlags(
            Intent.FLAG_ACTIVITY_NO_HISTORY or
                    Intent.FLAG_ACTIVITY_NEW_DOCUMENT or
                    Intent.FLAG_ACTIVITY_MULTIPLE_TASK
        )
        try {
            startActivity(intent)
        } catch (e: ActivityNotFoundException) {
            startActivity(
                Intent(
                    Intent.ACTION_VIEW,
                    Uri.parse("http://play.google.com/store/apps/details?id=" + this.packageName)
                )
            )
        }
    }

    private fun onGhClick() {
        logger.logMenuItemOpened("Github", sessionId.get())
        startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(Utils.URL_POLA_GH)))
    }

    private fun onFacebookClick() {
        logger.logMenuItemOpened("Pola na Facebooku", sessionId.get())
        startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(Utils.URL_POLA_FB)))
    }

    private fun onTwitterClick() {
        logger.logMenuItemOpened("Pola na Twitterze", sessionId.get())
        startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(Utils.URL_POLA_TWITTER)))
    }
}
